packages/common/src/ide/types/ide.types.ts

```diff
@@ -213,6 +213,12 @@ export interface IDE {
     editor: TextEditor,
     ranges: GeneralizedRange[],
   ): Promise<void>;
+
+  /**
+   * Write message to the Cursorless log
+   * @param message The message to log
+   */
+  log(message: string): Promise<void>;
```

Comment:
I'm hesitant to add more stuff to the top-level ide object.  As discussed, we'd like to break it up into smaller pieces.  Not strictly opposed to having it here, as it's just one function, but I'm not in love with this monolithic object that everyone needs

Also, do we want to encode some notion of log levels?  That way the places that we were using `console.debug` before could do something like one of the following

- `output.debug("foo")`, or
- `debug(output, "foo")`, or
- `output.log("foo", Log.debug)`

Reply:
I don't think there are different ways of logging things to the output channel, but you want this only to have some kind of log level I'm not against it. We already have a settings for the debug log.

Reply:
I think log level would be nice to have, and I would argue it's harder to add later, because we would need to go through and figure out what level everything should be